src/components/forms/ComboBox/ComboBox.test.tsx

```diff
@@ -0,0 +1,1111 @@
+    })
+
+    it('resets the filter on click', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      userEvent.type(getByTestId('combo-box-input'), 'b')
+      userEvent.click(getByTestId('combo-box-clear-button'))
+
+      expect(getByTestId('combo-box-clear-button')).not.toBeVisible()
+      expect(getByTestId('combo-box-option-list').children.length).toBe(
+        fruitOptions.length
+      )
+    })
+
+    it('calls onChange prop with undefined on click', () => {
+      const onChange = jest.fn()
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={onChange}
+        />
+      )
+
+      fireEvent.click(getByTestId('combo-box-clear-button'))
+
+      expect(onChange).toHaveBeenCalledWith(undefined)
+    })
+
+    it('clears the input on click', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      userEvent.type(getByTestId('combo-box-input'), 'b')
+      userEvent.click(getByTestId('combo-box-clear-button'))
+
+      expect(getByTestId('combo-box-clear-button')).not.toBeVisible()
+      expect(getByTestId('combo-box-input')).toHaveValue('')
+    })
+
+    it('works as expected when input is loaded with default value', () => {
+      const onChange = jest.fn()
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          defaultValue="apple"
+          onChange={onChange}
+        />
+      )
+      expect(getByTestId('combo-box-clear-button')).toBeVisible()
+
+      fireEvent.click(getByTestId('combo-box-clear-button'))
+
+      expect(getByTestId('combo-box-clear-button')).not.toBeVisible()
+      expect(getByTestId('combo-box-input')).toHaveValue('')
+      expect(onChange).toHaveBeenNthCalledWith(2, undefined)
+      expect(getByTestId('combo-box-option-list')).not.toBeVisible()
+      expect(getByTestId('combo-box-option-list').children.length).toBe(
+        fruitOptions.length
+      )
+    })
+
+    it('remains focused on the input after click', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      userEvent.type(getByTestId('combo-box-input'), 'b')
+      expect(getByTestId('combo-box-input')).toHaveFocus()
+
+      userEvent.click(getByTestId('combo-box-clear-button'))
+      fireEvent.blur(getByTestId('combo-box-clear-button'))
+      expect(getByTestId('combo-box-input')).toHaveFocus()
```

Comment:
should the clear button be tabbable? I wasn't able to in chrome

Reply:
Yes. This is documented in [a failing test](https://github.com/trussworks/react-uswds/blob/f6a08d3f59155e54c790f050c836a3826ff0c882/src/components/forms/ComboBox/ComboBox.test.tsx#L726-L746). Should it be a blocker? I'm thinking yes.